### PR TITLE
[post][ayi] Ayi a common util cli

### PR DESCRIPTION
**Pull request by at15**
Related issue: N/A

## Post info

- Title: Ayi 
- Author: @at15
- License: default
- Estimate finish time: ~~12/29 PST~~ 12/31 PST

## Related posts

- N/A

## Related projects

- [Ayi](https://github.com/dyweb/Ayi) A common util application

### CODE REVIEW

File: source/ayi.md [hunk 1]
```diff
@@ -10,9 +10,166 @@ preview: Ayi 跨平台的工具库
...
+
+- 跨平台 & 交叉编译
+- 简洁的包管理
+- 性能好，可以用来改进东岳现有的纯 PHP 服务端体系
+- 一个活跃的社区，PHP 沉浸在 CMS 和抄 Rails 中不能自拔，JS 日新月异
+- Google 老爹
+
+## 主要问题
+
+- 人太少，基本只有 @at15 (我) 一个人
```
Comment (gaocegege) [inline source/ayi.md:50]: 添加 GitHub 链接会好一些？

File: source/ayi.md [hunk 2]
```diff
@@ -10,9 +10,166 @@ preview: Ayi 跨平台的工具库
...
+  - 环境检查
+  - [git 操作的简化](https://github.com/dyweb/Ayi/tree/master/app/git)
+  - [makefile 类似的自动化工具](https://github.com/dyweb/Ayi/tree/master/util/runner)
+  - [静态 web 服务器](https://github.com/dyweb/Ayi/tree/master/app/web)
+  - [进程管理](https://github.com/dyweb/Ayi/pull/64)
+  - waka time 服务器
+  - 文件传输
+- go 的版本从 1.5 跳到了 1.7。原生支持 vendor 并且有了很多更好的依赖管理工具，比如 [glide](https://github.com/Masterminds/glide)
+
+第一个问题的话，基本无解，目前东岳经常写 Golang 的人好像只有我和策策。策策有空就要去陪妹子，自然不可能陪我来填坑。
```
Comment (gaocegege) [inline source/ayi.md:68]: 俗话说得好，万事靠自己

File: source/ayi.md [hunk 3]
```diff
@@ -10,9 +10,166 @@ preview: Ayi 跨平台的工具库
...
+
+- 咩在项目开始时提交了一些 shell 脚本，但是由于转到了 Golang 以及咩一向很忙，遂弃婶
+- @kdplus (丘) 参与过 `Ayi check` 的开发，不过那时我 Golang 菜的抠脚，导致丘也在划水。
+- @gaocegege (策策) 因为周报的功能，参与过一段时间的开发，
+引入了`Godep` 交叉编译，不过最后周报的功能并没有投入实用。
+
+
+## 总结
+
+- 等有钱了，给大家都配 MBP
```
Comment (gaocegege) [inline source/ayi.md:313]: 这个好

File: source/ayi.md [hunk 4]
```diff
@@ -10,9 +10,166 @@ preview: Ayi 跨平台的工具库
...
+
+
```
Comment (gaocegege) [inline source/ayi.md:315]: 多了一个换行

File: source/ayi.md (hunk 1 above)
Comment (at15) [reply source/ayi.md:50]: 懒...这个应该ink的render支持....

Comment (gaocegege) [reply source/ayi.md:50]: 是么，好吧，厉害了我的 ink

Comment (at15) [reply source/ayi.md:50]: no, I mean ... ink should have supported it...

Comment (gaocegege) [reply source/ayi.md:50]: 。。。。。。。。

Comment (JasonQSY) [reply source/ayi.md:50]: 可以mos开个坑=。= ink github支持插件=，=

File: source/ayi.md [hunk 5]
```diff
@@ -10,9 +10,166 @@ preview: Ayi 跨平台的工具库
...
+  - [git 操作的简化](https://github.com/dyweb/Ayi/tree/master/app/git)
+  - [makefile 类似的自动化工具](https://github.com/dyweb/Ayi/tree/master/util/runner)
+  - [静态 web 服务器](https://github.com/dyweb/Ayi/tree/master/app/web)
+  - [进程管理](https://github.com/dyweb/Ayi/pull/64)
+  - waka time 服务器
+  - 文件传输
+- go 的版本从 1.5 跳到了 1.7。原生支持 vendor 并且有了很多更好的依赖管理工具，比如 [glide](https://github.com/Masterminds/glide)
+
+第一个问题的话，基本无解，目前东岳经常写 Golang 的人好像只有我和策策。策策有空就要去陪妹子，自然不可能陪我来填坑。
+(要有妹子的话我还会去填坑么?)
```
Comment (JasonQSY) [inline source/ayi.md:69]: 单身还是第一生产力 👍

File: source/ayi.md [hunk 6]
```diff
@@ -10,9 +10,166 @@ preview: Ayi 跨平台的工具库
...
+
+- 等有钱了，给大家都配 MBP
+
+
+## 杂项
+
+- 使用 `git log -reverse` 可以反过来看log, 可以用来找第一个提交。
+- shell 在 windows 下基本不会有问题，因为为了使用 git，东岳所有的 windows 用户都安装了
+msysgit (现在叫 git for windows)，它自带了 bash 和一些基本的工具。
+- 周报的功能作为 MOS 的一个项目交给了 @codewarm96, 进度见[这个issue](https://github.com/dyweb/mos/issues/1)
```
Comment (JasonQSY) [inline source/ayi.md]: 拼错了=。= @codeworm96

**Review by JasonQSY — APPROVED** [review]
🐑

**Comment by JasonQSY** [comment]
感觉pull approve的体验不错。。（感觉我就是来玩pullapprove的

**Comment by gaocegege** [comment]
WIP 说吼是不管用的。。

**Comment by JasonQSY** [comment]
wip去掉以后重判嘛。。

**Comment by gaocegege** [comment]
好像是的

File: source/ayi.md (hunk 1 above)
Comment (arrowrowe) [reply source/ayi.md:50]: @JasonQSY 资磁~ 开个吧~

**Comment by arrowrowe** [comment]
:sheep: 

[Approved with PullApprove](https://pullapprove.com/dyweb/blog/pull-request/7/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=dyweb/blog)

File: source/ayi.md (hunk 1 above)
Comment (JasonQSY) [reply source/ayi.md:50]: go to https://github.com/dyweb/mos/issues/12

File: source/ayi.md (hunk 6 above)
Comment (ComMouse) [reply source/ayi.md]: @at15

File: source/ayi.md [hunk 7]
```diff
@@ -10,9 +10,166 @@ preview: Ayi 跨平台的工具库
...
+
+- 等有钱了，给大家都配 MBP
+
+
+## 杂项
+
+- 使用 `git log -reverse` 可以反过来看log, 可以用来找第一个提交。
+- shell 在 windows 下基本不会有问题，因为为了使用 git，东岳所有的 windows 用户都安装了
+msysgit (现在叫 git for windows)，它自带了 bash 和一些基本的工具。
+- 周报的功能作为 MOS 的一个项目交给了 @codeworm96, 进度见[这个issue](https://github.com/dyweb/mos/issues/1)
```
Comment (arrowrowe) [inline source/ayi.md:321]: fixed @at15 @ComMouse @JasonQSY

**Comment by ComMouse** [comment]
LGTM

[Approved with PullApprove](https://pullapprove.com/dyweb/blog/pull-request/7/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=dyweb/blog)

**Review by ComMouse — APPROVED** [review]

**Comment by at15** [comment]
I tried to add toc using https://github.com/jonschlinkert/markdown-toc, but ink is using https://github.com/russross/blackfriday which does not add `id` for `header`, so I would just give up for now..... 

It is possible to switch to https://github.com/shurcooL/github_flavored_markdown in ink and external toc generation would work

**Comment by JasonQSY** [comment]
LGTM

**Comment by JasonQSY** [comment]
好像还是没用到 blog group =。= @at15

**Comment by at15** [comment]
@JasonQSY seems to be a bug with pullapprove, collaborators are not synchronized.... I still see the old team members as potential reviewers in pullapprove's page ...

maybe create a issue here https://github.com/pullapprove/support ?

**Comment by gaocegege** [comment]
LGTM

[Approved with PullApprove](https://pullapprove.com/dyweb/blog/pull-request/7/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=dyweb/blog)

**Comment by arrowrowe** [comment]
:sheep: 

[Approved with PullApprove](https://pullapprove.com/dyweb/blog/pull-request/7/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=dyweb/blog)

**Comment by JasonQSY** [comment]
Know little about the configuration of pullapprove.. Maybe some one configuring it report the problem? @at15

File: source/ayi.md (hunk 2 above)
Comment (at15) [reply source/ayi.md:68]: 所以我们也不会去帮助 processing.R 的 /w

**Pull request merged**